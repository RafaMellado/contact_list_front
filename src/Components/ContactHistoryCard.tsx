import { Card } from "react-bootstrap";
import { useTranslation } from "react-i18next";
import { extractDateAndTimeFromDate } from "../Helpers/date";
import { ContactHistory } from "../Services/Interfaces/ContactHistory";

export interface ContactHistoryCardProps {
  item: ContactHistory;
  translations: string;
}

export function ContactHistoryCard({
  item,
  translations,
}: ContactHistoryCardProps) {
  const { t } = useTranslation();
  const { date, time } = extractDateAndTimeFromDate(item.created_at);

  return (
    <Card className="mb-3">
      <Card.Body>
        <h4>
          {t<string>(`${translations}.historyDay`, {
            date,
            time,
          })}
        </h4>
        <p>
          {item.givenname} {item.surname}
        </p>
        <p>{item.email}</p>
        <p>{item.phone}</p>
      </Card.Body>
    </Card>
  );
}
