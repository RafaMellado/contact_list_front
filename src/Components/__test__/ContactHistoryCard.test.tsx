import { ContactHistoryCard } from "../ContactHistoryCard";
import { act, render } from "@testing-library/react";
import type { ContactHistoryCardProps } from "../ContactHistoryCard";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (key: string) => key }),
}));

describe(`<${ContactHistoryCard.name} />`, () => {
  const contactHistoryCardProps: ContactHistoryCardProps = {
    item: {
      givenname: "First",
      surname: "Second",
      email: "Test email",
      phone: "Test phone",
      created_at: new Date().toString(),
    },
    translations: "contactHistory",
  };

  const factoryComponent = (
    props: ContactHistoryCardProps = contactHistoryCardProps
  ) => render(<ContactHistoryCard {...props} />);

  test("renders correctly the component", () => {
    const component = factoryComponent();

    act(() => {
      expect(component.asFragment()).toMatchSnapshot();
    });
  });
});
