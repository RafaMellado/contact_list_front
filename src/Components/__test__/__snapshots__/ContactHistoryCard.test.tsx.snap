// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactHistoryCard /> renders correctly the component 1`] = `
<DocumentFragment>
  <div
    class="mb-3 card"
  >
    <div
      class="card-body"
    >
      <h4>
        contactHistory.historyDay
      </h4>
      <small
        class="text-secondary"
      >
        contactHistory.name
      </small>
      <p>
        First Second
      </p>
      <small
        class="text-secondary"
      >
        contactHistory.email
      </small>
      <p>
        Test email
      </p>
      <small
        class="text-secondary"
      >
        contactHistory.email
      </small>
      <p>
        Test phone
      </p>
    </div>
  </div>
</DocumentFragment>
`;
